.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,48 @@
+name: Build and Push Docker Image
+
+on:
+  pull_request:
+    paths:
+      - Dockerfile
+    types:
+      - opened # default
+      - synchronize # default
+      - reopened # default
+      - ready_for_review # draft -> ready
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io/bebit
+  IMAGE_NAME: "${{ github.event_name == 'release' && 'python-mecab-builder' || 'python-mecab-builder-dev' }}"
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
```

Comment:
Can we use specific version instead of commit?
Also for other actions as well.

Reply:
Yeah that's the way I thought before, but checking the documentation from github they suggested using specific version/commit and it worked well with those versions
https://docs.github.com/en/actions/publishing-packages/publishing-docker-images

Reply:
```
# GitHub recommends pinning actions to a commit SHA.
```

Reply:
I got it, thank you.